app/controllers/contributions_controller.rb

```diff
@@ -45,10 +46,31 @@ def show
   end
 
   def edit
-    cho = EDM::ProvidedCHO.find_by(uuid: params[:uuid])
-    contribution = cho.edm_aggregatedCHO_for.contribution
+    contribution = contribution_from_params
     authorize! :edit, contribution
-    redirect_to send(:"edit_#{contribution.campaign.dc_identifier}_path", cho)
+    redirect_to send(:"edit_#{contribution.campaign.dc_identifier}_path", params[:uuid])
+  end
+
+  def delete
+    @contribution = contribution_from_params
+    authorize! :wipe, @contribution
+  end
+
+  def destroy
+    contribution = contribution_from_params
+    authorize! :wipe, contribution
+    begin
+      if contribution.first_published_at
```

Comment:
Could we add an `#ever_published?` method to `Contribution` which returns `true` if `first_published_at` is not `nil`, and use that here?